Yoast/wordpress-seo
# Gutenberg primary category picker

## Summary

This PR can be summarized in the following changelog entry:

* Adds support for selecting a primary taxonomy in Gutenberg.

## Relevant technical choices:

* Added a filter to `class-metabox.php`, `wpseo_content_meta_section_content` for adding additional hidden fields to the metabox.
* I opted to let the `PrimaryTaxonomyPicker` component manage it's own terms by calling the API, instead of using `wp.data.select( "core" )`, because this does not get updated when a new term is added in Gutenberg's taxonomy picker.

## Test instructions

This PR can be tested by following these steps:

* Install the `Custom Post Type UI` plugin. [Link](http://local.wordpress.test/wp-admin/plugin-install.php?tab=plugin-information&plugin=custom-post-type-ui&TB_iframe=true&width=600&height=550)
* To go `CPT UI` > `Tools` > `Taxonomies` and import the following:
```
{"hierarchical_post_taxonomy":{"name":"hierarchical_post_taxonomy","label":"Hierarchical Taxonomies","singular_label":"Hierarchical Taxonomy","description":"","public":"true","hierarchical":"true","show_ui":"true","show_in_menu":"true","show_in_nav_menus":"true","query_var":"true","query_var_slug":"","rewrite":"true","rewrite_slug":"","rewrite_withfront":"1","rewrite_hierarchical":"0","show_admin_column":"false","show_in_rest":"true","show_in_quick_edit":"","rest_base":"","labels":{"menu_name":"","all_items":"","edit_item":"","view_item":"","update_item":"","add_new_item":"","new_item_name":"","parent_item":"","parent_item_colon":"","search_items":"","popular_items":"","separate_items_with_commas":"","add_or_remove_items":"","choose_from_most_used":"","not_found":"","no_terms":"","items_list_navigation":"","items_list":""},"object_types":["post"]}}
```
* To verify the custom taxonomy works open the classic editor.

* In the classic editor, make sure primary taxonomies (Categories and "Hierarchical Taxonomies") work as intended.

## Quality assurance


Fixes #10601
